include/boost/geometry/algorithms/detail/distance/linear_or_areal_to_areal.hpp

```diff
@@ -41,16 +43,51 @@ struct linear_to_areal
                                     Areal const& areal,
                                     Strategy const& strategy)
     {
-        if ( geometry::intersects(linear, areal,
-                                  strategy.get_relate_segment_segment_strategy()) )
+        typedef typename point_type<Linear>::type point_type;
+
+        typedef segment_intersection_points<point_type> intersection_return_type;
+
+        intersection_return_type dis_res = geometry::detail::disjoint
+                       ::disjoint_with_info<Linear,Areal>
+                       ::apply(linear, areal,
+                               strategy.get_relate_segment_segment_strategy());
+
+        if ( dis_res.count > 0 )
         {
-            return 0;
+            return_type result;
+            strategy::distance::services::result_set_unique_point<Strategy>
+                    ::apply(result, dis_res.intersections[0]);
+            return result;
         }
 
-        return linear_to_linear
+        return_type res = linear_to_linear
             <
                 Linear, Areal, Strategy
             >::apply(linear, areal, strategy, false);
+
+        bool is_ring = boost::is_same
+                <
+                    typename tag<Areal>::type,
+                    ring_tag
+                >::type::value;
+
+        bool is_multi_polygon = boost::is_same
+                <
+                    typename tag<Areal>::type,
+                    multi_polygon_tag
+                >::type::value;
+
+        bool is_multi_linestring = boost::is_same
+                <
+                    typename tag<Linear>::type,
+                    multi_linestring_tag
+                >::type::value;
```

Comment:
Maybe it would be wise to implement them as utilities?
Then instead of:
```
boost::is_same
    <
        typename tag<Areal>::type,
        multi_polygon_tag
    >::type::value
```
you could write:
```
geometry::is_multi_polygon<Areal>::value
```

Reply:
good idea, done.